File: new/src/Logic/Connections/WarehouseConnection.cs
using System.Collections.Generic;

namespace AutoBuyer.Logic.Connections
{
    public class WarehouseConnection : IWarehouseConnection
    {
        private readonly List<StockItemConnection> _connections = List<StockItemConnection>();

        public IStockItemConnection ConnectToItem(String itemId)
        {
            var connection = StockItemConnection();
            _connections.Add(connection);

            return connection;
        }
    }
}

File: old/src/main/kotlin/logic/WarehouseConnection.kt
class WarehouseConnection : IWarehouseConnection {
    private val _connections = arrayListOf<StockItemConnection>();

    override fun connectToItem(itemId: String, buyerName: String): IStockItemConnection {
        var connection = StockItemConnection(buyerName);
        _connections.add(connection);

        return connection;
    }
}
